src/lib/operators/join_nested_loop.cpp

```diff
@@ -205,54 +207,60 @@ void JoinNestedLoop::_perform_join() {
                                           std::shared_ptr<PosList> pos_list) {
   // Add columns from table to output chunk
   for (ColumnID column_id{0}; column_id < input_table->column_count(); ++column_id) {
     std::shared_ptr<BaseColumn> column;
 
-    if (auto reference_column = std::dynamic_pointer_cast<const ReferenceColumn>(
-            input_table->get_chunk(ChunkID{0})->get_column(column_id))) {
-      auto new_pos_list = std::make_shared<PosList>();
-
-      ChunkID current_chunk_id{0};
+    if (input_table->type() == TableType::References) {
+      if (input_table->chunk_count() > 0) {
+        auto new_pos_list = std::make_shared<PosList>();
 
-      // de-reference to the correct RowID so the output can be used in a Multi Join
-      for (const auto row : *pos_list) {
-        if (row.chunk_id != current_chunk_id) {
-          current_chunk_id = row.chunk_id;
+        ChunkID current_chunk_id{0};
 
-          reference_column = std::dynamic_pointer_cast<const ReferenceColumn>(
-              input_table->get_chunk(current_chunk_id)->get_column(column_id));
-        }
-        if (row.chunk_offset == INVALID_CHUNK_OFFSET) {
-          new_pos_list->push_back(RowID{ChunkID{0}, INVALID_CHUNK_OFFSET});
-        } else {
-          new_pos_list->push_back(reference_column->pos_list()->at(row.chunk_offset));
+        // de-reference to the correct RowID so the output can be used in a Multi Join
+        for (const auto row : *pos_list) {
+          if (row.is_null()) {
+            new_pos_list->push_back(NULL_ROW_ID);
+          } else {
+            auto reference_column = std::static_pointer_cast<const ReferenceColumn>(
+                input_table->get_chunk(row.chunk_id)->get_column(column_id));
+            new_pos_list->push_back(reference_column->pos_list()->at(row.chunk_offset));
+          }
         }
-      }
 
-      column = std::make_shared<ReferenceColumn>(reference_column->referenced_table(),
-                                                 reference_column->referenced_column_id(), new_pos_list);
+        auto reference_column =
+            std::static_pointer_cast<const ReferenceColumn>(input_table->get_chunk(ChunkID{0})->get_column(column_id));
+
+        column = std::make_shared<ReferenceColumn>(reference_column->referenced_table(),
+                                                   reference_column->referenced_column_id(), new_pos_list);
+      } else {
+        // If there are no Chunks in the input_table, we can't deduce the Table that input_table is referencING to
+        // pos_list will contain only NULL_ROW_IDs anyway, so it doesn't matter which Table the ReferenceColumn that
+        // we output is referencing. HACK, but works fine: we create a dummy table and let the ReferenceColumn ref
+        // it.
+        const auto dummy_table = std::make_shared<Table>(input_table->column_definitions(), TableType::Data);
```

Comment:
Okay it is. How about adding a static method to `ReferenceColumn`: `create_with_dummy_table`?

Reply:
makes it just a tiny bit cleaner. but I went for it